modules/core/keeper/msg_server.go

```diff
@@ -764,8 +764,7 @@ func (k Keeper) ChannelUpgradeTry(goCtx context.Context, msg *channeltypes.MsgCh
 	if err != nil {
 		ctx.Logger().Error("channel upgrade try failed", "error", errorsmod.Wrap(err, "channel upgrade try failed"))
 		if channeltypes.IsUpgradeError(err) {
-			k.ChannelKeeper.MustAbortUpgrade(ctx, msg.PortId, msg.ChannelId, err)
-			cbs.OnChanUpgradeRestore(ctx, msg.PortId, msg.ChannelId)
+			_ = k.ChannelKeeper.WriteErrorReceipt(ctx, msg.PortId, msg.ChannelId, upgrade.Fields, err.(*channeltypes.UpgradeError))
```

Comment:
We ignore the error returned by `WriteErrorReceipt`, is that ok? Could we wrap it with the error returned by `ChanUpgradeTry` in case it happens so that we don't lose that information?

Reply:
we can possibly remove the error return later, the only error returned is if the channel is not found, which at this point is unreachable code